examples/hex-game/src/lib.rs

```diff
@@ -0,0 +1,247 @@
+        let player = self.active;
+        self.cell_mut(x, y).stone = Some(player);
+        self.active = player.other();
+        if !((x == 0 && player == Player::One)
+            || (y == 0 && player == Player::Two)
+            || self
+                .neighbors(x, y)
+                .any(|(nx, ny)| self.cell(nx, ny).is_connected(player)))
+        {
+            return;
+        }
+        let mut stack = vec![(x, y)];
+        while let Some((x, y)) = stack.pop() {
+            self.cell_mut(x, y).connected = true;
+            stack.extend(self.neighbors(x, y).filter(|(nx, ny)| {
+                let cell = self.cell(*nx, *ny);
+                !cell.connected && cell.stone == Some(player)
+            }));
+        }
+    }
+
+    /// Returns the winner, or `None` if the game is still in progress.
+    pub fn winner(&self) -> Option<Player> {
+        let s = self.size - 1;
+        for i in 0..self.size {
+            if self.cell(s, i).is_connected(Player::One) {
+                return Some(Player::One);
+            }
+            if self.cell(i, s).is_connected(Player::Two) {
+                return Some(Player::Two);
+            }
+        }
+        None
+    }
+
+    /// Returns the `Owner` controlling the active player.
+    pub fn active_player(&self) -> Player {
+        self.active
+    }
+
+    /// Returns the cell `(x, y)`.
+    fn cell(&self, x: u16, y: u16) -> Cell {
+        self.cells[x as usize + (y as usize) * (self.size as usize)]
+    }
+
+    /// Returns a mutable reference to cell `(x, y)`.
+    fn cell_mut(&mut self, x: u16, y: u16) -> &mut Cell {
+        &mut self.cells[x as usize + (y as usize) * (self.size as usize)]
+    }
+
+    /// Returns all neighbors of cell `(x, y)`, in the order:
+    /// left, top left, right, bottom right, top right, bottom left
+    fn neighbors(&self, x: u16, y: u16) -> impl Iterator<Item = (u16, u16)> {
+        iter::empty()
+            .chain((x > 0).then(|| (x - 1, y)))
+            .chain((y > 0).then(|| (x, y - 1)))
+            .chain((x + 1 < self.size).then(|| (x + 1, y)))
+            .chain((y + 1 < self.size).then(|| (x, y + 1)))
+            .chain((x + 1 < self.size && y > 0).then(|| (x + 1, y - 1)))
+            .chain((y + 1 < self.size && x > 0).then(|| (x - 1, y + 1)))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_simple_game() {
+        let mut board = Board::new(3);
+        // Make a few moves; no winner yet:
+        // 0 2 2
+        //  2 1 0
+        //   1 0 1
+        assert_eq!(Player::One, board.active_player());
+        assert_eq!(MoveOutcome::Ok, board.make_move(0, 2));
+        assert_eq!(Player::Two, board.active_player());
+        assert_eq!(MoveOutcome::Ok, board.make_move(0, 1));
+        assert_eq!(MoveOutcome::Ok, board.make_move(1, 1));
+        assert_eq!(MoveOutcome::Ok, board.make_move(1, 0));
+        assert_eq!(MoveOutcome::Ok, board.make_move(2, 2));
```

Comment:
Shouldn't player one win here?

Reply:
No, because `(1, 1)` is not a neighbor of `(2, 2)`. (It _is_ a neighbor of `(0, 2)`, though.)

I tried to illustrate the game situation in the comment above.

Reply:
Oh right! Of course :man_facepalming: . Somehow I thought that `(1,1)` and `2,2` are neighbours